bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/handler/EnOceanClassicDeviceHandler.java

```diff
@@ -276,13 +276,12 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     @Override
     public void handleRemoval() {
-        ScheduledFuture<?> future = releaseFuture;
-        if (future != null && !future.isDone()) {
-            future.cancel(true);
-            future = null;
+        ScheduledFuture<?> releaseFuture = this.releaseFuture;
```

Comment:
Please revert to as it was, only change one line:

```suggestion
            future.cancel(true);
            releaseFuture = null;
        }

        super.handleRemoval();
```

Reply:
No, on thing removal the job should be cancelled unconditionally.